spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -158,30 +185,39 @@ protected void onInit() {
 					.getBean(IntegrationContextUtils.ARGUMENT_RESOLVER_MESSAGE_CONVERTER_BEAN_NAME,
 							SmartMessageConverter.class));
 		}
+		if (clientManager != null) {
+			clientManager.addCallback(this);
```

Comment:
Cannot we do this in a single place in the `super.onInit()` ?

Reply:
Good idea!